src/services/dumpers/agent-nodejs.ts

```diff
@@ -191,7 +202,38 @@ export default class AgentNodeJs extends AbstractDumper {
     });
   }
 
-  protected createFiles(dumpConfig: Config) {
+  private writeModels(schema) {
+    const collectionNamesSorted = Object.keys(schema).sort();
+
+    collectionNamesSorted.forEach(collectionName => {
+      const { fields, options } = schema[collectionName];
+      const modelPath = `models/primary/${this.lodash.kebabCase(collectionName)}.js`;
+
+      const fieldsDefinition = fields.map(field => {
+        return {
+          ...field,
+          ref: field.ref && this.strings.transformToCamelCaseSafeString(field.ref),
+        };
+      });
+
+      this.copyHandleBarsTemplate(`models/model.hbs`, modelPath, {
+        modelName: this.strings.transformToCamelCaseSafeString(collectionName),
+        collectionName,
+        fields: fieldsDefinition,
+        timestamps: options.timestamps,
+      });
+    });
+  }
+
+  private async writeMongooseModels(schema) {
+    await this.mkdirp(`${this.projectPath}/models/primary`);
+
+    this.copyHandleBarsTemplate('models/index.hbs', 'models/primary/index.js');
+
+    this.writeModels(schema);
+  }
```

Comment:
I think we should merge both function.

`writeModels` is mongoose specific atm, and `writeMongooseModels` does almost nothing outside creating the folder and copy index.

Reply:
Ok let's go.